lib/grandpa/grandpa.go

```diff
@@ -24,34 +24,112 @@ import (
 
 // Service represents the current state of the grandpa protocol
 type Service struct {
-	state         *State // current state
-	blockState    BlockState
-	subround      subround                           // current sub-round
-	votes         map[ed25519.PublicKeyBytes]*Vote   // votes for next state
-	equivocations map[ed25519.PublicKeyBytes][]*Vote // equivocatory votes for this stage
-	head          common.Hash                        // most recently finalized block hash
+	// preliminaries
+	blockState BlockState
+	keypair    *ed25519.Keypair //nolint
```

Comment:
?

Reply:
I started working on the next algo which needs a keypair, I just left it in for now since it'll be used in the next PR, but can remove if preferred

Reply:
fine as is, just curious